pkg/lifecyclehooks/lyfecyclehooks.go

```diff
@@ -0,0 +1,207 @@
+package lifecyclehooks
```

Comment:
Wondering if we need a separate package for this or whether we should just move it into `infra`, WDYT?